sktime/forecasting/compose/_pipeline.py

```diff
@@ -1462,6 +1488,23 @@ def _get_forecaster_X_prediction(self, X=None, fh=None, method="predict"):
             return None
         if isinstance(self.columns, (list, pd.Index)) and len(self.columns) == 0:
             return X
+
+        # if user passes data for future unknown variables, do not forecast them
+        # this is done only if predict_behaviour is "use_actuals"
+        if isinstance(self.columns, (list, pd.Index)) and len(self.columns):
+            is_future_unknown_actually_known = all(
+                column in X.columns for column in self.columns
+            )
```

Comment:
If user has specifically declared what all features are future-unknown using `columns`, I am checking here if all of those are present in passed `X` argument or not. If even one of them is missing, `forecaster_X` must be used for predicting them in future.